psutil/_psutil_sunos.c

```diff
@@ -116,6 +116,58 @@ psutil_proc_basic_info(PyObject *self, PyObject *args) {
         );
 }
 
+/*
+ * Join array of C strings to C string with delemiter dm.
+ * Omit empty records.
+ */
+static int
+cstrings_array_to_string(char **joined, char ** array, size_t count, char dm) {
+    int i;
+    size_t total_length = 0;
+    size_t item_length = 0;
+    char *result = NULL;
+    char *last = NULL;
+
+    if (!array || !joined)
+        return 0;
```

Comment:
You should set a proper Python exception here. Later on, when you invoke this function, you arbitrarily `goto error` but I think that this way it will generate a void "python exception not set" error. I think something like this should be good:
```
PyErr_SetString(PyExc_ValueError, "array size mismatch");
```